File: live2vod.js
// ==================== 完全动态爬虫（零硬编码）+ 合集模式 ====================
// 版本: 11.1.0
// 特性: 所有数据源均来自 ext 配置，支持集合播放列表

const header = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
};

// ==================== 全局变量 ====================
let dynamicClasses = [];
let extBasePath = "";
let defaultBasePath = "https://raw.githubusercontent.com/mannys888/frist/refs/heads/main/";
let cache = {};
let debugMode = true;

// ==================== 日志 ====================
function log(msg, level = "INFO") {
    if (!debugMode && level === "DEBUG") return;
    console.log(`[${level}] ${msg}`);
}

// ==================== 网络请求（带缓存） ====================
function fetchSync(url, useCache = true) {
    if (useCache && cache[url]) return cache[url];
    try {
        log(`请求: ${url}`, "DEBUG");
        let response = req(url, { method: 'GET', headers: header });
        let content = (typeof response === 'string') ? response : (response?.content || null);
        if (content && useCache) cache[url] = content;
        return content;
    } catch (e) {
        log(`请求失败: ${url} - ${e.message}`, "ERROR");
        return null;
    }
}

// ==================== 工具函数 ====================
function getRandomCover(title) {
    let hash = 0;
    for (let i = 0; i < (title || "").length; i++) hash = ((hash << 5) - hash) + title.charCodeAt(i);
    let themes = ["music", "nature", "abstract", "art", "film"];
    let theme = themes[Math.abs(hash) % themes.length];
    return `https://picsum.photos/200/300?random=${Math.abs(hash) % 100}&theme=${theme}`;
}

function getFileType(url) {
    if (!url) return "📄 未知";
    let ext = url.split('.').pop().toLowerCase();
    let types = {
        'mp3': '🎵 音频', 'wav': '🎵 音频', 'ogg': '🎵 音频', 'flac': '🎵 音频',
        'mp4': '🎬 视频', 'mkv': '🎬 视频', 'avi': '🎬 视频', 'mov': '🎬 视频',
        'm3u8': '📺 直播', 'flv': '📺 直播', 'ts': '📺 直播',
        'jpg': '🖼️ 图片', 'png': '🖼️ 图片', 'gif': '🖼️ 图片',
        'txt': '📄 文本', 'json': '📋 数据', 'xml': '📋 数据'
    };
    return types[ext] || '🎵 媒体';
}

function resolvePath(path, basePath) {
    if (!path) return "";
    if (path.match(/^https?:\/\//i)) return path;
    if (path.startsWith('data:')) return path;
    let base = basePath.endsWith('/') ? basePath : basePath + '/';
    if (path.startsWith('./')) return base + path.substring(2);
    if (path.startsWith('../')) {
        let parts = base.split('/');
        let upCount = (path.match(/\.\.\//g) || []).length;
        for (let i = 0; i < upCount && parts.length > 3; i++) parts.pop();
        let newBase = parts.join('/') + '/';
        return newBase + path.replace(/\.\.\//g, '');
    }
    if (path.startsWith('/')) {
        let match = base.match(/^(https?:\/\/[^/]+)/);
        return match ? match[1] + path : base + path.substring(1);
    }
    return base + path;
}

function parseContent(content, options = {}) {
    let items = [];
    let separators = options.separators || [',', '\t', '|', '$'];
    let lines = content.split(/\r?\n/);
    for (let line of lines) {
        if (!line || line.trim() === "") continue;
        if (line.startsWith('#') || line.startsWith('//')) continue;
        let foundSep = null, sepIdx = -1;
        for (let sep of separators) {
            let idx = line.indexOf(sep);
            if (idx > 0 && (sepIdx === -1 || idx < sepIdx)) { sepIdx = idx; foundSep = sep; }
        }
        if (foundSep) {
            let title = line.substring(0, sepIdx).trim();
            let url = line.substring(sepIdx + 1).trim();
            items.push({ title, url });
        } else if (line.match(/^https?:\/\//i)) {
            items.push({ title: "媒体文件", url: line });
        }
    }
    return items;
}

// ==================== ext 配置解析 ====================
function parseExtConfig(extParam, basePath) {
    let classes = [];
    if (!extParam) return classes;

    try {
        let configData = null;
        if (typeof extParam === 'string' && extParam.match(/^https?:\/\//i)) {
            log(`下载 ext 配置: ${extParam}`, "INFO");
            let content = fetchSync(extParam);
            if (content) {
                try {
                    configData = JSON.parse(content);
                } catch(e) { configData = content; }
            }
        } 
        else if (typeof extParam === 'string' && (extParam.trim().startsWith('{') || extParam.trim().startsWith('['))) {
            try { configData = JSON.parse(extParam); } catch(e) { configData = extParam; }
        }
        else if (typeof extParam === 'object') {
            configData = extParam;
        }
        else if (typeof extParam === 'string') {
            configData = extParam;
        }

        if (configData) {
            if (Array.isArray(configData)) {
                for (let item of configData) {
                    if (item.name) {
                        let typeId = item.url || item.typeId || item.id || item.name;
                        if (typeId && !typeId.match(/^https?:\/\//i)) {
                            typeId = resolvePath(typeId, basePath);
                        }
                        classes.push({
                            type_name: item.name,
                            type_id: typeId,
                            icon: item.icon || "",
                            description: item.description || "",
                            handler: item.handler || null,
                            parseConfig: item.parseConfig || null
                        });
                    }
                }
            }
            else if (configData.sites && Array.isArray(configData.sites)) {
                for (let site of configData.sites) {
                    if (site.name) {
                        let typeId = site.url || site.api || site.key || site.name;
                        if (typeId && !typeId.match(/^https?:\/\//i)) {
                            typeId = resolvePath(typeId, basePath);
                        }
                        classes.push({
                            type_name: site.name,
                            type_id: typeId,
                            icon: site.icon || "",
                            handler: site.handler || null,
                            parseConfig: site.parseConfig || null
                        });
                    }
                }
            }
            else if (configData.categories && Array.isArray(configData.categories)) {
                for (let cat of configData.categories) {
                    if (cat.name) {
                        let typeId = cat.url || cat.id || cat.name;
                        if (typeId && !typeId.match(/^https?:\/\//i)) {
                            typeId = resolvePath(typeId, basePath);
                        }
                        classes.push({
                            type_name: cat.name,
                            type_id: typeId,
                            icon: cat.icon || "",
                            handler: cat.handler || null,
                            parseConfig: cat.parseConfig || null
                        });
                    }
                }
            }
            else if (typeof configData === 'string') {
                let lines = configData.split(/\r?\n/);
                for (let line of lines) {
                    if (!line.trim()) continue;
                    let parts = line.split(',');
                    if (parts.length >= 2) {
                        classes.push({
                            type_name: parts[0].trim(),
                            type_id: resolvePath(parts[1].trim(), basePath),
                            handler: null
                        });
                    } else if (parts.length === 1 && parts[0].match(/^https?:\/\//i)) {
                        classes.push({
                            type_name: "链接 " + (classes.length + 1),
                            type_id: parts[0].trim(),
                            handler: null
                        });
                    }
                }
            }
        }
    } catch(e) {
        log(`解析 ext 失败: ${e.message}`, "ERROR");
    }

    if (classes.length === 0) {
        log("警告: ext 配置未生成任何分类，请检查配置内容", "WARN");
    }
    return classes;
}

// ==================== 处理器（如 cctv） ====================
function handleCategoryByHandler(handler, tid, pg, context) {
    if (handler === "cctv") {
        let cctvChannels = context.cctvChannels || [];
        if (cctvChannels.length === 0) {
            log("未在 ext 中配置 cctvChannels，返回空", "WARN");
            return [];
        }
        let videos = [];
        for (let ch of cctvChannels) {
            videos.push({
                vod_id: ch.id + "###cctv",
                vod_name: ch.name,
                vod_pic: getRandomCover(ch.name),
                vod_remarks: "📺 直播"
            });
        }
        return videos;
    }
    if (typeof handler === 'function') {
        return handler(tid, pg, context);
    }
    return null;
}

// ==================== 文件处理器（支持合集模式） ====================
function handleFileSource(fileUrl, parseConfig, basePath, asCollection = false) {
    let resolvedUrl = fileUrl;
    if (!resolvedUrl.match(/^https?:\/\//i)) {
        resolvedUrl = resolvePath(fileUrl, basePath);
    }
    if (parseConfig && parseConfig.autoExt && !resolvedUrl.includes('.')) {
        let testUrl = resolvedUrl + parseConfig.autoExt;
        let testContent = fetchSync(testUrl, true);
        if (testContent) {
            resolvedUrl = testUrl;
            log(`自动补全扩展名: ${resolvedUrl}`, "INFO");
        }
    }
    log(`请求文件: ${resolvedUrl}`, "INFO");
    let content = fetchSync(resolvedUrl);
    if (!content) return [];

    let items = [];
    if (parseConfig && parseConfig.type === "json") {
        try {
            let json = JSON.parse(content);
            let dataArr = parseConfig.dataPath ? json[parseConfig.dataPath] : (Array.isArray(json) ? json : (json.list || []));
            for (let item of dataArr) {
                let title = parseConfig.titleField ? item[parseConfig.titleField] : (item.title || item.name);
                let url = parseConfig.urlField ? item[parseConfig.urlField] : (item.url || item.link);
                if (title && url) items.push({ title, url });
            }
        } catch(e) { log("JSON解析失败", "ERROR"); }
    } 
    else if (parseConfig && parseConfig.type === "regex") {
        let regex = new RegExp(parseConfig.pattern, parseConfig.flags || 'g');
        let match;
        while ((match = regex.exec(content)) !== null) {
            let title = match[parseConfig.titleGroup] || "未命名";
            let url = match[parseConfig.urlGroup];
            if (url) items.push({ title, url });
        }
    }
    else {
        items = parseContent(content, { separators: parseConfig?.separators || [',', '\t', '|', '$'] });
    }

    // 合集模式：返回单个条目，vod_id 后缀为 ###file
    if (asCollection && items.length > 0) {
        let collectionName = parseConfig?.collectionName || (fileUrl.split('/').pop().replace(/\.(txt|m3u8?|json)$/i, '') + " 合集");
        let vod_id = resolvedUrl + "###file";
        return [{
            vod_id: vod_id,
            vod_name: collectionName,
            vod_pic: getRandomCover(collectionName),
            vod_remarks: `共${items.length}集`
        }];
    }

    // 非合集模式：每个资源一个条目
    let videos = [];
    let fileBase = resolvedUrl.substring(0, resolvedUrl.lastIndexOf('/') + 1);
    for (let item of items) {
        if (item.url) {
            let link = item.url.match(/^https?:\/\//i) ? item.url : resolvePath(item.url, fileBase);
            videos.push({
                vod_id: link + "###music",
                vod_name: item.title || "未命名",
                vod_pic: getRandomCover(item.title),
                vod_remarks: getFileType(link)
            });
        }
    }
    return videos;
}

// ==================== 对外接口 ====================
function init(extend) {
    log(`========== 爬虫初始化（合集模式支持） ==========`, "INFO");
    extBasePath = defaultBasePath;
    if (extend && typeof extend === 'string' && extend.match(/^https?:\/\//i)) {
        extBasePath = extend.substring(0, extend.lastIndexOf('/') + 1);
    }
    dynamicClasses = parseExtConfig(extend, extBasePath);
    log(`生成 ${dynamicClasses.length} 个分类`, "INFO");
}

function home() {
    return JSON.stringify({ class: dynamicClasses, filters: null });
}

function homeVod() {
    return JSON.stringify({ list: [] });
}

function category(tid, pg, filter, extend) {
    pg = parseInt(pg) || 1;
    log(`category: tid=${tid}, pg=${pg}`, "DEBUG");
    if (pg >= 2) {
        return JSON.stringify({ list: [], page: pg, pagecount: 1, limit: 90, total: 0 });
    }

    let classConfig = dynamicClasses.find(c => c.type_id === tid || c.type_name === tid);
    if (!classConfig) {
        log(`未找到分类配置: ${tid}`, "WARN");
        return JSON.stringify({ list: [], page: pg, pagecount: 0, total: 0 });
    }

    let videos = [];
    let handler = classConfig.handler;
    let parseConfig = classConfig.parseConfig || {};
    let collectionMode = parseConfig.collectionMode === true;

    if (handler) {
        let context = { 
            cctvChannels: parseConfig.cctvChannels || [], 
            basePath: extBasePath,
            tid, pg 
        };
        let result = handleCategoryByHandler(handler, tid, pg, context);
        if (result) videos = result;
        else log(`处理器 ${handler} 未返回数据`, "WARN");
    }

    if (videos.length === 0) {
        let fileUrl = classConfig.type_id;
        videos = handleFileSource(fileUrl, parseConfig, extBasePath, collectionMode);
    }

    return JSON.stringify({
        list: videos,
        page: pg,
        pagecount: 1,
        limit: 90,
        total: videos.length
    });
}

function detail(vodId) {
    log(`detail: ${vodId}`, "DEBUG");
    let parts = vodId.split('###');
    if (parts.length < 2) return JSON.stringify({ list: [] });
    let videoId = parts[0];
    let type = parts[1];

    // 合集类型（文件集合）
    if (type === "file") {
        let fileUrl = videoId;
        if (!fileUrl.match(/^https?:\/\//i)) fileUrl = resolvePath(fileUrl, extBasePath);
        let content = fetchSync(fileUrl);
        if (!content) return JSON.stringify({ list: [] });
        let baseDir = fileUrl.substring(0, fileUrl.lastIndexOf('/') + 1);
        let items = [];

        // JSON
        if (content.trim().startsWith('{') || content.trim().startsWith('[')) {
            try {
                let json = JSON.parse(content);
                let arr = Array.isArray(json) ? json : (json.list || json.data || []);
                for (let item of arr) {
                    let title = item.title || item.name || "未命名";
                    let url = item.url || item.link || item.src || item.play_url;
                    if (url) {
                        if (!url.match(/^https?:\/\//i)) url = resolvePath(url, baseDir);
                        items.push(`${title}$${url}`);
                    }
                }
            } catch(e) {}
        }
        // M3U
        if (items.length === 0 && content.includes("#EXTM3U")) {
            let lines = content.split(/\r?\n/);
            let currentTitle = "";
            for (let line of lines) {
                line = line.trim();
                if (line.startsWith("#EXTINF:")) {
                    let match = line.match(/#EXTINF:.*?,(.*)/);
                    if (match) currentTitle = match[1].trim();
                } else if (line && !line.startsWith("#")) {
                    if (line.match(/^https?:\/\//i)) {
                        items.push(`${currentTitle || "直播流"}$${line}`);
                        currentTitle = "";
                    }
                }
            }
        }
        // 通用分隔符
        if (items.length === 0) {
            let separators = [',', '\t', '|', '$'];
            let parsed = parseContent(content, { separators });
            for (let p of parsed) {
                items.push(`${p.title}$${p.url}`);
            }
        }

        if (items.length === 0) return JSON.stringify({ list: [] });
        let playUrl = items.join("#");
        let firstTitle = items[0].split('$')[0] || "媒体合集";
        let vod = {
            vod_id: vodId,
            vod_name: firstTitle,
            vod_pic: getRandomCover(firstTitle),
            vod_play_from: "播放列表",
            vod_play_url: playUrl
        };
        return JSON.stringify({ list: [vod] });
    }

    // 原有 cctv 直播
    if (type === "cctv") {
        let classConfig = dynamicClasses.find(c => c.handler === "cctv");
        let channels = classConfig?.parseConfig?.cctvChannels || [];
        let channel = channels.find(ch => ch.id === videoId);
        let streamUrl = channel ? channel.url : "";
        let vod = {
            vod_id: videoId,
            vod_name: channel ? channel.name : "央视直播",
            vod_pic: getRandomCover("cctv"),
            vod_play_from: "央视直播",
            vod_play_url: streamUrl ? ("直播流$" + streamUrl) : ""
        };
        return JSON.stringify({ list: [vod] });
    } else {
        let title = decodeURIComponent(videoId.split('/').pop().split('.')[0] || "媒体播放");
        let vod = {
            vod_id: videoId,
            vod_name: title,
            vod_pic: getRandomCover(title),
            vod_play_from: "播放源",
            vod_play_url: "播放$" + videoId
        };
        return JSON.stringify({ list: [vod] });
    }
}

function play(flag, id, vipFlags) {
    log(`play: ${id}`, "DEBUG");
    return JSON.stringify({ parse: 0, url: id });
}

function search(keyword, page) {
    return JSON.stringify({ list: [] });
}

__JS_SPIDER__ = { init, home, homeVod, category, detail, play, search };

File: live2vod.py
# coding=utf-8
#!/usr/bin/python
import sys
sys.path.append('..')
from base.spider import Spider
import json
import time
import base64
import re
import traceback

class Spider(Spider):
    def getName(self):
        return "小学学习"

    def init(self, extend=""):
        print("============ 小学学习爬虫初始化 ============")
        print("extend参数: %s" % extend)
        self.category_url = "https://raw.githubusercontent.com/mannys888/frist/refs/heads/main/cateManual(1).json"
        # 文件内容缓存，避免重复下载
        self.file_cache = {}

    def isVideoFormat(self, url):
        pass

    def manualVideoCheck(self):
        pass

    def homeContent(self, filter):
        result = {}
        classes = []
        try:
            rsp = self.fetch(self.category_url, headers=self.header)
            if rsp.status_code == 200:
                cateManual = rsp.json()
                for k in cateManual:
                    classes.append({
                        'type_name': k,
                        'type_id': cateManual[k]
                    })
            else:
                raise Exception("HTTP %d" % rsp.status_code)
        except Exception as e:
            print("获取分类配置失败: %s" % e)
            print("使用默认硬编码分类")
            default_categories = {
                "迦南诗歌": "迦南诗歌.txt",
                "经典咏流传": "yypy.txt",
                "赞美诗歌": "zm.txt"
            }
            for k in default_categories:
                classes.append({
                    'type_name': k,
                    'type_id': default_categories[k]
                })
        result['class'] = classes
        result['filters'] = {} if filter else None
        return result

    def homeVideoContent(self):
        return {'list': []}

    def _parse_file_to_videos(self, file_url):
        """解析文本文件，返回视频列表（每个元素为 {title, url}）"""
        if file_url in self.file_cache:
            return self.file_cache[file_url]
        try:
            rsp = self.fetch(file_url, headers=self.header)
            if rsp.status_code != 200:
                print("获取文件失败: %s" % rsp.status_code)
                return []
            content = rsp.text
            videos = []
            lines = content.splitlines()
            for line in lines:
                line = line.strip()
                if not line or line.startswith('#'):
                    continue
                title = None
                link = None
                # 支持分隔符: , | \t 空格
                for sep in [',', '|', '\t', ' ']:
                    if sep in line:
                        parts = line.split(sep, 1)
                        if len(parts) == 2:
                            title = parts[0].strip()
                            link = parts[1].strip()
                            break
                if not title and re.match(r'^https?://', line):
                    title = "媒体文件"
                    link = line
                if title and link:
                    videos.append({"title": title, "url": link})
            self.file_cache[file_url] = videos
            return videos
        except Exception as e:
            print("解析文件出错: %s" % e)
            traceback.print_exc()
            return []

    def categoryContent(self, tid, pg, filter, extend):
        result = {}
        pg = int(pg) if pg else 1
        print("categoryContent: tid=%s, pg=%d" % (tid, pg))

        # 央视分类分支（保留）
        if tid.startswith('TOPC'):
            filterParams = ["id", "p", "d"]
            params = []
            for fp in filterParams:
                if fp in extend:
                    params.append("{0}={1}".format(fp, extend[fp]))
                elif fp == 'id':
                    params.append("id={0}".format(tid))
            suffix = '&'.join(params)
            url = 'https://api.cntv.cn/NewVideo/getVideoListByColumn?{0}&n=20&sort=desc&mode=0&serviceId=tvcctv&t=json'.format(suffix)
            print("央视请求URL: %s" % url)
            try:
                rsp = self.fetch(url, headers=self.header)
                jo = json.loads(rsp.text)
                vodList = jo['data']['list']
                videos = []
                for vod in vodList:
                    guid = vod['guid']
                    title = vod['title']
                    img = vod['image']
                    videos.append({
                        "vod_id": guid + "###" + img,
                        "vod_name": title,
                        "vod_pic": img,
                        "vod_remarks": ''
                    })
                result['list'] = videos
                result['page'] = pg
                result['pagecount'] = 9999
                result['limit'] = 90
                result['total'] = 999999
            except Exception as e:
                print("央视接口请求失败: %s" % e)
                result['list'] = []
                result['page'] = pg
                result['pagecount'] = 0
                result['limit'] = 90
                result['total'] = 0
            return result

        # 普通文件分类：每个 txt 文件作为一个合集（单条目），提高用户体验
        file_url = "https://raw.githubusercontent.com/mannys888/frist/refs/heads/main/{0}".format(tid)
        videos = self._parse_file_to_videos(file_url)
        if not videos:
            result['list'] = []
        else:
            # 构造一个虚拟的 vod_id，用于 detailContent 识别（格式：file://文件URL###分类名）
            # 同时存储分类名，便于显示
            vod_id = "file://{0}###{1}".format(file_url, tid)
            result['list'] = [{
                "vod_id": vod_id,
                "vod_name": tid.replace('.txt', '') + " 合集",   # 显示为“迦南诗歌 合集”
                "vod_pic": "",   # 可以设置一个默认封面
                "vod_remarks": "共{}集".format(len(videos))
            }]
        result['page'] = pg
        result['pagecount'] = 1
        result['limit'] = 20
        result['total'] = len(videos)
        return result

    def detailContent(self, array):
        vod_input = array[0]
        # 处理文件合集类型
        if vod_input.startswith('file://'):
            # 格式：file://文件URL###分类名
            parts = vod_input.split('###')
            if len(parts) >= 2:
                file_url = parts[0][7:]  # 去掉 "file://"
                category_name = parts[1]
                videos = self._parse_file_to_videos(file_url)
                if videos:
                    # 拼接播放列表：标题1$url1#标题2$url2
                    play_url = "#".join(["{0}${1}".format(v['title'], v['url']) for v in videos])
                    vod = {
                        "vod_id": vod_input,
                        "vod_name": category_name.replace('.txt', '') + " 合集",
                        "vod_pic": "",
                        "type_name": "教育",
                        "vod_year": "",
                        "vod_area": "",
                        "vod_remarks": "共{}集".format(len(videos)),
                        "vod_actor": "",
                        "vod_director": "",
                        "vod_content": "本合集包含{}个视频，请在播放列表中选择。".format(len(videos)),
                        "vod_play_from": "直链",
                        "vod_play_url": play_url
                    }
                    return {'list': [vod]}
                else:
                    return {'list': []}

        # 直链类型（旧格式：url###single）
        if vod_input.startswith('http') and '###single' in vod_input:
            url_part = vod_input.split('###')[0]
            title = url_part.split('/')[-1].split('.')[0]
            vod = {
                "vod_id": vod_input,
                "vod_name": title,
                "vod_pic": "",
                "type_name": '',
                "vod_year": "",
                "vod_area": "",
                "vod_remarks": "",
                "vod_actor": "",
                "vod_director": "",
                "vod_content": "",
                "vod_play_from": "直链",
                "vod_play_url": title + "$" + url_part
            }
            return {'list': [vod]}

        # 央视 pid 类型（pid###img）
        if '###' in vod_input:
            parts = vod_input.split('###')
            pid = parts[0]
            pic_url = parts[1] if len(parts) > 1 else ""
            url = "https://vdn.apps.cntv.cn/api/getHttpVideoInfo.do?pid={0}".format(pid)
            try:
                rsp = self.fetch(url, headers=self.header)
                jo = json.loads(rsp.text)
                title = jo.get('title', '未知标题').strip()
                link = jo.get('hls_url', '').strip()
                if not link:
                    raise Exception("没有获取到播放地址")
                vod = {
                    "vod_id": pid,
                    "vod_name": title,
                    "vod_pic": pic_url,
                    "type_name": '',
                    "vod_year": "",
                    "vod_area": "",
                    "vod_remarks": "",
                    "vod_actor": "",
                    "vod_director": "",
                    "vod_content": "",
                    "vod_play_from": "央视源",
                    "vod_play_url": title + "$" + link
                }
                return {'list': [vod]}
            except Exception as e:
                print("获取详情失败: %s" % e)
                vod = {
                    "vod_id": pid,
                    "vod_name": "播放失败",
                    "vod_pic": pic_url,
                    "vod_play_from": "错误",
                    "vod_play_url": ""
                }
                return {'list': [vod]}

        # 保底：直接当作普通链接处理
        vod = {
            "vod_id": vod_input,
            "vod_name": "未知视频",
            "vod_pic": "",
            "vod_play_from": "直链",
            "vod_play_url": "视频$" + vod_input
        }
        return {'list': [vod]}

    def searchContent(self, key, quick):
        return {'list': []}

    def playerContent(self, flag, url, vipFlags):
        result = {}
        result["parse"] = 0
        result["playUrl"] = ''
        result["url"] = url
        result["header"] = {
            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36"
        }
        return result

    config = {
        "player": {},
        "filter": {}
    }

    header = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.54 Safari/537.36",
        "Referer": "https://www.cntv.cn"
    }

    def localProxy(self, param):
        return [200, "video/MP2T", "", ""]